tools/sqlite3_api_wrapper/sqlite3_api_wrapper.cpp

```diff
@@ -215,7 +215,7 @@ int sqlite3_prepare_v2(sqlite3 *db,           /* Database handle */
 
 		// extract the remainder of the query and assign it to the pzTail
 		if (pzTail && set_remainder) {
-			*pzTail = zSql + next_location + 1;
+			*pzTail = zSql + next_location;
```

Comment:
`next_location` is obtained from the statement that is returned (see the code above). If this was the problem I would imagine multiple statements on a line would also not work in native DuckDB. I think the issue here might actually be with the statements returned by MD not setting `stmt_length` correctly. Could you verify that that is being set correctly?

Reply:
You're absolutely right! The `stmt_length` was off by one actaully and it should be fixed now. PTAL. Thanks!